Fixes #8290 - dry up foreman_url logic in one place

dustints opened the pull request:
depends on https://github.com/theforeman/foreman/pull/1902

Commit:
Fixes #8290 - dry up foreman_url logic in one place

ares commented:
Looks good, could you please add unit tests for `Foreman::Controller::ForemanUrlRenderable`?

ares commented on app/controllers/concerns/foreman/controller/foreman_url_renderable.rb:
```diff
@@ -0,0 +1,50 @@
+      foreman_url_from_config_or_request(action, host.token)
+    end
+  end
+
```
everything below this line should be private methods I think

ares commented:
Also could you please rebase the PR, #1902 was merged so you don't have to include the commit here (it causes conflicts)

ares commented:
Meanwhile I tested the patch, works nicely (in both config template rendering and TFTP file rendering).

dustints commented:
@ares thanks for testing and reviewing.
Updated
- rebased
- added unit test for foreman_url_renderable, added in #8291 but forgot to add it here.
- made most methods private in foreman_url_renderable

dustints commented:
mysql builds failing because of jenkins config issue

```
14:37:43 ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
14:37:43 hudson.AbortException: No test report files were found. Configuration error?
```

orrabin commented:
there is a known mysql error: http://projects.theforeman.org/issues/8537
if that's the only error ignore it for now and test again after the issue is fixed

orrabin commented:
[test]

orrabin commented:
@dustints can you please rebase?
